### PR TITLE
List command

### LINKED ISSUE #30: mortar list command to list all shots

### PR DESCRIPTION
**Pull request by jnummelin**
fixes #30 

```
$  bundle exec mortar list
SHOT            RESOURCES
hello            10
```
`RESOURCES` refers to the count of resources fund from the given shot

```
$  bundle exec mortar list -q
hello
```

Also quiet option to allow easy scripting around.

### CODE REVIEW

File: lib/mortar/list_command.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        next unless shot_name # Kube API gives back some ComponentStatus resources not actually matching
+
+        if shots[shot_name]
+          shots[shot_name] += 1
+        else
+          shots[shot_name] = 1
+        end
+      end
+
+      tabs = "\t\t\t"
```
Comment (kke) [inline lib/mortar/list_command.rb]: The describe PR brings in TTY::Table so maybe use it here :)

Comment (jnummelin) [reply lib/mortar/list_command.rb]: yeah, that was the idea :)

Comment (jnummelin) [reply lib/mortar/list_command.rb]: done

**Comment by jnummelin** [comment]
@kke added completions and changed to use tty-table. also added some docs for this, plus docs for missing describe command. :) PTAL

File: lib/mortar/list_command.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+module Mortar
+  class ListCommand < Mortar::Command
+    include Mortar::ClientHelper
+    include Mortar::TTYHelper
+
+    option ['-q', '--quiet'], :flag, "only output shot names"
+
+    def execute
+      shots = {}
```
Comment (kke) [inline lib/mortar/list_command.rb]: with `shots = Hash.new(0)` you can change the 

```ruby
        if shots[shot_name]
          shots[shot_name] += 1
        else
          shots[shot_name] = 1
        end
```
to just `shots[shot_name] += 1`

File: lib/mortar/root_command.rb [hunk 3]
```diff
@@ -20,6 +21,7 @@ class RootCommand < Clamp::Command
     subcommand "fire", "Fire a shot (of k8s manifests)", FireCommand
     subcommand "yank", "Yank a shot (of k8s manifests)", YankCommand
     subcommand "describe", "Describe a shot (of k8s manifests)", DescribeCommand
+    subcommand "list", "List a shots (of k8s manifests)", ListCommand
```
Comment (kke) [inline lib/mortar/root_command.rb]: Extra `a` or `s`.

File: opt/bash-completion.sh [hunk 4]
```diff
@@ -40,6 +40,11 @@ _mortar () {
           COMPREPLY=( $(compgen -W "table yaml json" -- ${cur}) )
         fi
         ;;
+      list)
+        if [[ ${cur} == -* ]] ; then
+          COMPREPLY=( $(compgen -W "--debug --help" -- ${cur}) )
```
Comment (kke) [inline opt/bash-completion.sh]: `--quiet` missing

File: lib/mortar/list_command.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+        if shots[shot_name]
+          shots[shot_name] += 1
+        else
+          shots[shot_name] = 1
+        end
+      end
+
+      if quiet?
+        shots.keys.each do |k|
```
Comment (kke) [inline lib/mortar/list_command.rb]: can be `shots.each_key`

File: lib/mortar/list_command.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          puts k
+        end
+      else
+        table = TTY::Table.new %w(NAME RESOURCES), []
```
Comment (kke) [inline lib/mortar/list_command.rb]:
```suggestion
  table = TTY::Table.new %w(NAME RESOURCES), shots.to_a
```

(can omit the loop below)

File: opt/bash-completion.sh (hunk 4 above)
Comment (kke) [reply opt/bash-completion.sh]:
```suggestion
          COMPREPLY=( $(compgen -W "--quiet --debug --help" -- ${cur}) )
```

File: lib/mortar/root_command.rb (hunk 3 above)
Comment (kke) [reply lib/mortar/root_command.rb]:
```suggestion
    subcommand "list", "List shots (of k8s manifests)", ListCommand
```

File: lib/mortar/list_command.rb (hunk 5 above)
Comment (kke) [reply lib/mortar/list_command.rb]:
```suggestion
        shots.each_key do |k|
```

File: lib/mortar/list_command.rb (hunk 2 above)
Comment (jnummelin) [reply lib/mortar/list_command.rb]: what black magic is that?
```
2.4.2 :003 > {} == Hash.new(0)
 => true 
```

**Comment by jnummelin** [comment]
@kke fixed all the comments, PTAL

**Pull request merged**